src/errors/mod.rs

```diff
@@ -0,0 +1,234 @@
+pub mod io;
+
+extern crate url;
+
+use crate::cloud_provider::Kind;
+use url::Url;
+
+pub struct SimpleError {
+    message: String,
+    message_safe: String,
+}
+
+#[derive(Debug)]
+pub enum LogLevel {
+    Debug,
+    Info,
+    Warning,
+    Error,
+    Critical,
+}
+
+#[derive(Debug)]
+pub enum Stage {
+    Infrastructure(InfrastructureStep),
+    Environment(EnvironmentStep),
+}
+
+#[derive(Debug)]
+pub enum InfrastructureStep {
+    Instantiate,
+    Create,
+    Upgrade,
+    Delete,
+}
+
+#[derive(Debug)]
+pub enum EnvironmentStep {
+    Build,
+    Deploy,
+    Update,
+    Delete,
```

Comment:
Pause ?

Reply:
Sure ! Thanks, I wasn't exhaustive, this list will grow along migration :)